pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -319,377 +201,317 @@ def _get_updateinfo_xml_path(root_elem):
         return data_elems[0].find(xpath).get("href")
 
 
-class CoreMetadataTestCase(PulpTestCase):
-    """Publish repository and validate the metadata is the same.
+@pytest.fixture(scope="class")
+def assert_created_publication(init_and_sync, rpm_publication_api, monitor_task):
+    def _assert_created_publication(url):
+        """Sync and publish an RPM repository given a feed URL."""
+        repo, _ = init_and_sync(url=url)
+        publish_data = RpmRpmPublication(repository=repo.pulp_href)
+        publish_response = rpm_publication_api.create(publish_data)
+        assert monitor_task(publish_response.task).created_resources != []
 
-    This Test does the following:
+    return _assert_created_publication
 
-    1. Create a rpm repo and a remote.
-    2. Sync the repo with the remote.
-    3. Publish and distribute the repo.
-    4. Download the metadata of both the original and Pulp-created repos.
-    5. Convert it into a canonical form (XML -> JSON -> Dict -> apply a sorting order).
-    6. Compare the metadata.
-    """
 
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        cls.cfg = config.get_config()
-        cls.client = gen_rpm_client()
-        cls.repo_api = RepositoriesRpmApi(cls.client)
-        cls.remote_api = RemotesRpmApi(cls.client)
-        cls.publications = PublicationsRpmApi(cls.client)
-        cls.distributions = DistributionsRpmApi(cls.client)
-
-    def setUp(self):
-        """Setup to run before every test."""
-        delete_orphans()
-
-    def test_complex_repo(self):
-        """Test the "complex" fixture that covers more of the metadata cases.
-
-        The standard fixtures have no changelogs and don't cover "ghost" files. The repo
-        with the "complex-package" does, and also does a better job of covering rich deps
-        and other atypical metadata.
-        """
-        self.do_metadata_comparison_test(RPM_COMPLEX_FIXTURE_URL)
+@pytest.mark.parallel
+def test_publish_rpm_rich_weak(assert_created_publication):
+    """Sync and publish an RPM rich weak repository."""
+    assert_created_publication(RPM_RICH_WEAK_FIXTURE_URL)
 
-    # TODO: make generic enough to apply against other repos
-    # (with multiple RPMs that could be published in a different order)
-    def do_metadata_comparison_test(self, repo_url):
```

Comment:
These comments went missing too

Reply:
This method was initially a helper method for a single test in a class. I merged the helper method and the test itself to `test_complex_repo_core_metadata`.